### PR TITLE
Keep phone numbers as strings

### PR DESCRIPTION
**Pull request by dcheno**
Meant to address issues where phone numbers with leading `+`s or `0`s were getting cast to integers.

Manually tested by @michaellevy and passes existing unit tests. I think it would be a good idea to add some additional tests for this specific bug, but I don't have a ton of spare hours at the moment to add them. So posting as-is for now.

@michaellevy also added synthetic data with `+`s or `0`s

### CODE REVIEW

File: cider/featurizer.py [hunk 1]
```diff
@@ -288,9 +289,7 @@ def write_bc(index: Any, iterator: Any) -> Any:
                                             ('reporting' not in col) or (col == 'reporting__number_of_records')])
         cdr_features = cdr_features.toDF(*[c if c == 'name' else 'cdr_' + c for c in cdr_features.columns])
         save_df(cdr_features, self.outputs_path / 'datasets' / 'bandicoot_features' / 'all.csv')
-        self.features['cdr'] = read_csv(
-            self.spark, self.outputs_path / 'datasets' / 'bandicoot_features' / 'all.csv', header=True, inferSchema=True
-        )
+        self.features['cdr'] = cdr_features
```
Comment (dcheno) [inline cider/featurizer.py:292]: did this in a few places where we were writing out and then reading back. Assumes that the original dataframe we wrote out has the columns the way we want them, which seemed accurate from doing manual testing.

Comment (michaellevy) [reply cider/featurizer.py:292]: @emilylaiken -- do you remember if there was a reason for this write-then-read idiom?

Comment (emilylaiken) [reply cider/featurizer.py:292]: This seems fine. iirc the only reason to read back in is something related to modularity (i.e. if there's a crash, you can always start by reading in the the features). That made more sense back when it was research code; I think here it's fine to assume one successful run from start to finish.

File: cider/featurizer.py [hunk 2]
```diff
@@ -561,9 +551,20 @@ def load_features(self) -> None:
...
                 try:
-                    self.features[feature] = read_csv(
-                        self.spark, data_path / f'{path_to_dataset}.csv', header=True, inferSchema=True
-                    )
+                    # reading through pandas to prevent phone numbers being read as integers without either having to specify
+                    # the whole schema or having to read twice with spark.
+                    # Alternatives if this ends up being too slow:
+                    #  - use pyspark.pandas (would require spark 3.2 upgrde)
+                    #  - do a limited read, grab schema, edit and re-read
+                    pandas_df = pd.read_csv(data_path / f'{path_to_dataset}.csv', dtype={'caller_id': 'str', 'name': 'str'})
```
[the remark this answers is not in the document]
Comment (dcheno) [reply cider/featurizer.py:558]: Good catch, updated.

**Comment by michaellevy** [comment]
Merging per @dcheno 's request...

**Pull request merged**